### ISSUE TITLE
Bug when using custom columns to do filters for the past 1 day/month/year

### ISSUE THREAD

**Issue by Jerrylovescoding** [metabase/metabase#37165]
### Metabase Version: 
v0.48.1

### Describe Bug:
When I use custom datetime column to filter last 1 month, I get a wrong result. Viewing the sql, I find the converting sql is wrong.
![image](https://github.com/metabase/metabase/assets/25549512/d0c1537e-2086-46a1-b893-3a7b12583b02)

### To Reproduce

1. Go to query editor
![image](https://github.com/metabase/metabase/assets/25549512/946b8d02-b64e-49a2-a7ec-9739577faa3d)

2. Add a new custom datetime column
![image](https://github.com/metabase/metabase/assets/25549512/d1cf739d-88f6-43a2-946b-53c97370354a)

3. Use custom column as a filter
![image](https://github.com/metabase/metabase/assets/25549512/27e370d0-4bef-4211-bc90-1a39038c124d)

4. View the wrong sql
![image](https://github.com/metabase/metabase/assets/25549512/9083c2b5-4f94-47e2-a454-0c6587f58759)

### Information about your Metabase installation

```JSON
"metabase-info": {
    "databases": [
      "mysql",
      "postgres",
      "mongo",
      "clickhouse"
    ],
    "hosting-env": "docker",
    "application-database": "postgres",
    "application-database-details": {
      "database": {
        "name": "PostgreSQL",
        "version": "13.9"
      },
      "jdbc-driver": {
        "name": "PostgreSQL JDBC Driver",
        "version": "42.6.0"
      }
    }
 }
```

### Severity

We got a wrong result if we use query editor in that way.

**Comment by darksciencebase** [comment]
I suspect this was fixed in https://github.com/metabase/metabase/pull/35523 which we had to revert because other things got broken. The [new approach to fixing it](https://github.com/metabase/metabase/pull/36068) is in review, I'll check if that one fixes this particular case

**Comment by darksciencebase** [comment]
ok, it's not fixed there, we'll have to look into it.

**Comment by lbrdnk** [comment]
Issue was fixed by PRs #38886 and #38874. It is not reproducible with those PRs merged.
